### PR TITLE
Create dao layer

### PR DESCRIPTION
**Pull request by legafise**

### CODE REVIEW

File: src/main/java/com/lashkevich/stores/dao/impl/GoodPriceDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    private static final String UPDATE_GOOD_PRICE_SQL = "UPDATE good_prices SET country_id = ?, good_id = ?, price = ? WHERE country_id = ? AND good_id = ?;";
+    private static final String DELETE_PRICE_BY_COUNTRY_AND_GOOD_SQL = "DELETE FROM baskets WHERE country_id = ? AND good_id = ?;";
+
+    @Override
+    public void add(GoodPrice goodPrice) throws DaoStoreException {
+        try (Connection connection = ConnectionUtil.getConnection();
+             PreparedStatement preparedStatement = connection.prepareStatement(ADD_TO_GOOD_PRICE_SQL)) {
+            preparedStatement.setLong(1, goodPrice.getCountry());
+            preparedStatement.setLong(2, goodPrice.getGood());
+            preparedStatement.setDouble(2, goodPrice.getPrice());
```
Comment (IvanYarochkin) [inline src/main/java/com/lashkevich/stores/dao/impl/GoodPriceDaoImpl.java]: Shouldn't be the same number `2` for these lines.

File: src/main/java/com/lashkevich/stores/entity/User.java [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (o == this){
+            return true;
+        }
+
+        if(o == null || getClass() != o.getClass()) {
+            return true;
```
[the remark this answers is not in the document]
Comment (legafise) [reply src/main/java/com/lashkevich/stores/entity/User.java]: Done

File: src/main/java/com/lashkevich/stores/entity/GoodPrice.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
+package com.lashkevich.stores.entity;
+
+import java.util.Objects;
+
+public class GoodPrice {
+    private long country;
+    private long good;
+    private double price;
```
[the remark this answers is not in the document]
Comment (legafise) [reply src/main/java/com/lashkevich/stores/entity/GoodPrice.java]: Done

File: src/main/java/com/lashkevich/stores/entity/GoodPrice.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
+package com.lashkevich.stores.entity;
+
+import java.util.Objects;
+
+public class GoodPrice {
+    private long country;
+    private long good;
```
[the remark this answers is not in the document]
Comment (legafise) [reply src/main/java/com/lashkevich/stores/entity/GoodPrice.java]: Done

File: src/main/java/com/lashkevich/stores/entity/GoodPrice.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
+package com.lashkevich.stores.entity;
+
+import java.util.Objects;
+
+public class GoodPrice {
+    private long country;
```
[the remark this answers is not in the document]
Comment (legafise) [reply src/main/java/com/lashkevich/stores/entity/GoodPrice.java]: Done

File: src/main/java/com/lashkevich/stores/entity/City.java [hunk 6]
```diff
@@ -0,0 +1,66 @@
+package com.lashkevich.stores.entity;
+
+import java.util.Objects;
+
+public class City {
+    private long id;
+    private String name;
+    private long country;
```
[the remark this answers is not in the document]
Comment (legafise) [reply src/main/java/com/lashkevich/stores/entity/City.java]: Done

File: src/main/java/com/lashkevich/stores/entity/Basket.java [hunk 7]
```diff
@@ -0,0 +1,54 @@
+package com.lashkevich.stores.entity;
+
+import java.util.Objects;
+
+public class Basket {
+    private long good;
```
[the remark this answers is not in the document]
Comment (legafise) [reply src/main/java/com/lashkevich/stores/entity/Basket.java]: Done

File: src/main/java/com/lashkevich/stores/entity/Basket.java [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+    private long user;
```
[the remark this answers is not in the document]
Comment (legafise) [reply src/main/java/com/lashkevich/stores/entity/Basket.java]: Done

File: src/main/java/com/lashkevich/stores/entity/User.java [hunk 9]
```diff
@@ -0,0 +1,154 @@
...
+
+public class User {
+    private long id;
+    private String name;
+    private String surname;
+    private String login;
+    private String password;
+    private String email;
+    private LocalDate birthDate;
+    private long role;
```
[the remark this answers is not in the document]
Comment (legafise) [reply src/main/java/com/lashkevich/stores/entity/User.java]: Done

File: src/main/java/com/lashkevich/stores/entity/User.java [hunk 10]
```diff
@@ -0,0 +1,154 @@
...
+    private long city;
```
[the remark this answers is not in the document]
Comment (legafise) [reply src/main/java/com/lashkevich/stores/entity/User.java]: Done

File: src/main/java/com/lashkevich/stores/exception/ConnectionStoreException.java [hunk 11]
```diff
@@ -0,0 +1,22 @@
+package com.lashkevich.stores.exception;
+
+public class ConnectionStoreException extends Exception{
```
[the remark this answers is not in the document]
Comment (legafise) [reply src/main/java/com/lashkevich/stores/exception/ConnectionStoreException.java]: Done

File: src/main/java/com/lashkevich/stores/util/converter/DateConverter.java [hunk 12]
```diff
@@ -0,0 +1,14 @@
+package com.lashkevich.stores.util.converter;
+
+import java.sql.Date;
+import java.time.LocalDate;
+
+public class DateConverter {
+    public static java.sql.Date convertLocalDateToDate(LocalDate localDate) {
+        return java.sql.Date.valueOf(localDate);
```
[the remark this answers is not in the document]
Comment (legafise) [reply src/main/java/com/lashkevich/stores/util/converter/DateConverter.java]: Done

File: src/main/java/com/lashkevich/stores/dao/BasketDao.java [hunk 13]
```diff
@@ -0,0 +1,18 @@
...
+
+import com.lashkevich.stores.entity.Basket;
+import com.lashkevich.stores.exception.DaoStoreException;
+
+import java.util.List;
+
+public interface BasketDao {
+    void add(Basket basket) throws DaoStoreException;
+
+    List<Basket> getAll() throws DaoStoreException;
```
[the remark this answers is not in the document]
Comment (legafise) [reply src/main/java/com/lashkevich/stores/dao/BasketDao.java]: Done

File: src/main/java/com/lashkevich/stores/dao/BasketDao.java [hunk 14]
```diff
@@ -0,0 +1,18 @@
...
+
+    Basket getByUser(long id) throws DaoStoreException;
```
[the remark this answers is not in the document]
Comment (legafise) [reply src/main/java/com/lashkevich/stores/dao/BasketDao.java]: Done

File: src/main/java/com/lashkevich/stores/dao/CityDao.java [hunk 15]
```diff
@@ -0,0 +1,18 @@
...
+
+import com.lashkevich.stores.entity.City;
+import com.lashkevich.stores.exception.DaoStoreException;
+
+import java.util.List;
+
+public interface CityDao {
+    void add(City city) throws DaoStoreException;
+
+    List<City> getAll() throws DaoStoreException;
```
[the remark this answers is not in the document]
Comment (legafise) [reply src/main/java/com/lashkevich/stores/dao/CityDao.java]: Done

File: src/main/java/com/lashkevich/stores/dao/CityDao.java [hunk 16]
```diff
@@ -0,0 +1,18 @@
...
+
+    City getById(long id) throws DaoStoreException;
```
[the remark this answers is not in the document]
Comment (legafise) [reply src/main/java/com/lashkevich/stores/dao/CityDao.java]: Done

File: src/main/java/com/lashkevich/stores/dao/CountryDao.java [hunk 17]
```diff
@@ -0,0 +1,18 @@
...
+
+import com.lashkevich.stores.entity.Country;
+import com.lashkevich.stores.exception.DaoStoreException;
+
+import java.util.List;
+
+public interface CountryDao {
+    void add(Country country) throws DaoStoreException;
+
+    List<Country> getAll() throws DaoStoreException;
```
[the remark this answers is not in the document]
Comment (legafise) [reply src/main/java/com/lashkevich/stores/dao/CountryDao.java]: Done

File: src/main/java/com/lashkevich/stores/dao/CountryDao.java [hunk 18]
```diff
@@ -0,0 +1,18 @@
...
+
+    Country getById(long id) throws DaoStoreException;
```
[the remark this answers is not in the document]
Comment (legafise) [reply src/main/java/com/lashkevich/stores/dao/CountryDao.java]: Done

File: src/main/java/com/lashkevich/stores/dao/GoodDao.java [hunk 19]
```diff
@@ -0,0 +1,18 @@
...
+
+import com.lashkevich.stores.entity.Good;
+import com.lashkevich.stores.exception.DaoStoreException;
+
+import java.util.List;
+
+public interface GoodDao {
+    void add(Good good) throws DaoStoreException;
+
+    List<Good> getAll() throws DaoStoreException;
```
[the remark this answers is not in the document]
Comment (legafise) [reply src/main/java/com/lashkevich/stores/dao/GoodDao.java]: Done

File: src/main/java/com/lashkevich/stores/dao/GoodDao.java [hunk 20]
```diff
@@ -0,0 +1,18 @@
...
+
+    Good getById(long id) throws DaoStoreException;
+
+    void update(Good good) throws DaoStoreException;
```
[the remark this answers is not in the document]
Comment (legafise) [reply src/main/java/com/lashkevich/stores/dao/GoodDao.java]: Done

File: src/main/java/com/lashkevich/stores/dao/GoodPriceDao.java [hunk 21]
```diff
@@ -0,0 +1,18 @@
...
+
+import com.lashkevich.stores.entity.GoodPrice;
+import com.lashkevich.stores.exception.DaoStoreException;
+
+import java.util.List;
+
+public interface GoodPriceDao {
+    void add(GoodPrice goodPrice) throws DaoStoreException;
+
+    List<GoodPrice> getAll() throws DaoStoreException;
```
[the remark this answers is not in the document]
Comment (legafise) [reply src/main/java/com/lashkevich/stores/dao/GoodPriceDao.java]: Done

File: src/main/java/com/lashkevich/stores/dao/GoodPriceDao.java [hunk 22]
```diff
@@ -0,0 +1,18 @@
...
+
+    GoodPrice getByCountryAndGood(long countryId, long goodId) throws DaoStoreException;
```
[the remark this answers is not in the document]
Comment (legafise) [reply src/main/java/com/lashkevich/stores/dao/GoodPriceDao.java]: Done

File: src/main/java/com/lashkevich/stores/dao/RoleDao.java [hunk 23]
```diff
@@ -0,0 +1,18 @@
...
+
+import com.lashkevich.stores.entity.Role;
+import com.lashkevich.stores.exception.DaoStoreException;
+
+import java.util.List;
+
+public interface RoleDao {
+    void add(Role role) throws DaoStoreException;
+
+    List<Role> getAll() throws DaoStoreException;
```
[the remark this answers is not in the document]
Comment (legafise) [reply src/main/java/com/lashkevich/stores/dao/RoleDao.java]: Done

File: src/main/java/com/lashkevich/stores/dao/RoleDao.java [hunk 24]
```diff
@@ -0,0 +1,18 @@
...
+
+    Role getById(long id) throws DaoStoreException;
```
[the remark this answers is not in the document]
Comment (legafise) [reply src/main/java/com/lashkevich/stores/dao/RoleDao.java]: Done

File: src/main/java/com/lashkevich/stores/dao/UserDao.java [hunk 25]
```diff
@@ -0,0 +1,18 @@
...
+
+import com.lashkevich.stores.entity.User;
+import com.lashkevich.stores.exception.DaoStoreException;
+
+import java.util.List;
+
+public interface UserDao {
+    void add(User user) throws DaoStoreException;
+
+    List<User> getAll() throws DaoStoreException;
```
[the remark this answers is not in the document]
Comment (legafise) [reply src/main/java/com/lashkevich/stores/dao/UserDao.java]: Done

File: src/main/java/com/lashkevich/stores/dao/UserDao.java [hunk 26]
```diff
@@ -0,0 +1,18 @@
...
+
+    User getById(long id) throws DaoStoreException;
```
[the remark this answers is not in the document]
Comment (legafise) [reply src/main/java/com/lashkevich/stores/dao/UserDao.java]: Done

File: src/main/java/com/lashkevich/stores/dao/impl/GoodPriceDaoImpl.java (hunk 1 above)
Comment (legafise) [reply src/main/java/com/lashkevich/stores/dao/impl/GoodPriceDaoImpl.java]: Done

File: src/test/java/com/lashkevich/stores/test/BasketDaoTest.java [hunk 27]
```diff
@@ -0,0 +1,86 @@
...
+    public void addTest() throws DaoStoreException {
+        Assert.assertTrue(basketDao.add(firstExpectedBasket,10));
+    }
+
+    @Test
+    public void removeTest() throws DaoStoreException {
+        Assert.assertTrue(basketDao.remove(9));
+
+    }@Test
+    public void updateTest() throws DaoStoreException {
```
Comment (IvanYarochkin) [inline src/test/java/com/lashkevich/stores/test/BasketDaoTest.java:83]: Format this code

File: src/test/java/com/lashkevich/stores/test/CityDaoTest.java [hunk 28]
```diff
@@ -0,0 +1,65 @@
+package com.lashkevich.stores.test;
```
Comment (IvanYarochkin) [inline src/test/java/com/lashkevich/stores/test/CityDaoTest.java:1]: Use package as for CityDao (com.lashkevich.stores.dao)

**Pull request merged**